src/net/mod.rs

```diff
@@ -1654,6 +1664,8 @@ pub enum HttpResponseType {
     ServerError(HttpResponseMetadata, String),
     ServiceUnavailable(HttpResponseMetadata, String),
     Error(HttpResponseMetadata, u16, String),
+    GetHealthError(HttpResponseMetadata, serde_json::Value),
```

Comment:
Yes, I thought of doing that. 

The reason I didn't is because the function that generates the responses is generic (`parse_error`). 

```
let mut error_text = String::new();
        let mut json_val: serde_json::Value = Default::default();
        fd.read_to_string(&mut error_text)
            .map_err(net_error::ReadError)?;
        if preamble.content_type == HttpContentType::JSON {
            let json_result = serde_json::from_str(&error_text.clone());
            json_val = match json_result {
                Ok(val) => val,
                Err(serde_json::Error { .. }) => {
                    return Err(net_error::DeserializeError(
                        "Invalid response: unable to deserialize json response".to_string(),
                    ))
                }
            }
        }

        let md = HttpResponseMetadata::from_preamble(request_version, preamble);
        let resp = match preamble.status_code {
            400 => HttpResponseType::BadRequest(md, error_text),
            401 => HttpResponseType::Unauthorized(md, error_text),
            402 => HttpResponseType::PaymentRequired(md, error_text),
            403 => HttpResponseType::Forbidden(md, error_text),
            404 => HttpResponseType::NotFound(md, error_text),
            500 => HttpResponseType::ServerError(md, error_text),
            503 => HttpResponseType::ServiceUnavailable(md, error_text),
            512 => HttpResponseType::GetHealthError(md, json_val),
            513 => HttpResponseType::GetHealthNoDataError(md, error_text),
            _ => HttpResponseType::Error(md, preamble.status_code, error_text),
        };
        Ok(resp)
```

It seemed like better style not to match specifically on `HttpResponseType::GetHealthError`. I see the argument for doing that though, let me know what you think.

Reply:
That makes sense to me. If we end up with more json objects, then we can refactor this later.